plasmapy/physics/parameters.py

```diff
@@ -399,18 +399,112 @@ def thermal_speed(T, particle="e", method="most_probable"):
 
     # different methods, as per https://en.wikipedia.org/wiki/Thermal_velocity
     if method == "most_probable":
-        V = (np.sqrt(2*k_B*T/m)).to(units.m/units.s)
+        V = (np.sqrt(2 * k_B * T / m)).to(units.m / units.s)
     elif method == "rms":
-        V = (np.sqrt(3*k_B*T/m)).to(units.m/units.s)
+        V = (np.sqrt(3 * k_B * T / m)).to(units.m / units.s)
     elif method == "mean_magnitude":
-        V = (np.sqrt(8*k_B*T/(m/np.pi))).to(units.m/units.s)
+        V = (np.sqrt(8 * k_B * T / (m / np.pi))).to(units.m / units.s)
     else:
         raise(ValueError("Method {} not supported in thermal_speed"
                          .format(method)))
 
     return V
 
 
+@check_relativistic
+@check_quantity({
+    'T': {'units': units.K, 'can_be_negative': False}
+})
+def kappa_thermal_speed(T, kappa, particle="e", method="most_probable"):
```

Comment:
I like the option with the warning, depending on how often the kappa distribution (heard nothing about it) is needed for electrons.

Reply:
I second the warning method. Is PhysicsWarning ready yet?

Reply:
My current thinking is we should make this a new issue, default to electrons, and then implement the warning using a physics specific warning based on #168  and #170  once that pull request is completed. Then make sure such defaults are implemented throughout plasmapy (as we assumes particles in many functions). Thoughts?

Reply:
Creating another issue for this does sound like the best idea since it is much broader than this pull request.  A possibility for the moment would be to put in a `UserWarning` that can be changed to a `PhysicsWarning` or something more specific later on, if you'd like.  But, it'd be fine with me to wait to resolve this with the later issue.